hammer_time/hammer_time.py

```diff
@@ -140,6 +141,34 @@ def perform(cls, client, machine_id):
         ctrl_client.juju('ssh', (machine_id,) + cls.kill_script)
 
 
+class InterruptNetworkAction(MachineAction):
+
+    def get_command():
+        deny_all = '; '.join([
+            'iptables --flush',
+            'iptables -P FORWARD DROP',
+            'iptables -P OUTPUT DROP',
+            'iptables -P INPUT DROP',
+            ])
+        restore = 'iptables-restore < $HOME/iptables'
+        commands = [
+            'set -eux',
+            'sudo iptables-save > $HOME/iptables',
+            'echo "{}"| sudo at now + 6 minutes'.format(restore),
+            'echo "{}"| sudo at now + 1 minutes'.format(deny_all),
+            ]
+        return '; '.join(commands)
```

Comment:
I contemplated something similar with the proxy tests I wrote, but I chose a different path.

I think there is a very rare chance that restore will fail and if InterruptNetworkAction is called again, it could save the bad rules. well maybe not because nothing can get in to instrument another test. So if this fails, we cannot even get information to do an autopsy. If this is a concern, maybe we want to ensure we can always ssh in
    iptables -A INPUT -p tcp --dport 22 -j ACCEPT